Spicy Opal Woodpecker

medium

# angouropoutanos69-making external calls while looping it is extremely dangerous because if it reverts it can cause DoS

angouropoutanos69
Medium
# angouropoutanos69-making external calls while looping it is extremely dangerous because if it reverts it can cause DoS
## Summary
The problem is in the function `multicall` there is a loop with an external call. Making external calls while looping them is dangerous because if it runs out of gas or reverts it can cause DoS, especially if the loop index is user controlled wich in this case is. Anyone could put the loop index very high and and make the function run out of gas. 
## Vulnerability Detail
For more information about this vulnerability 
https://github.com/sherlock-audit/2023-12-truflation/blob/main/truflation-contracts/src/token/TrufVesting.sol#L540
For more information about this vulnerability see secureum101 `43.Calls inside a loop`
## Impact
DoS
## Code Snippet
       (bool success, bytes memory result) = address(this).delegatecall(payloads[i]);
## Tool used

Manual Review

## Recommendation
Do not use external calls in loops. Moreover if you have to make sure that it is bounded not allowing the user access to the loop index. 